core/src/Kokkos_TypeInfo.hpp

```diff
@@ -0,0 +1,92 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef KOKKOS_TYPE_INFO_HPP
+#define KOKKOS_TYPE_INFO_HPP
+
+#include <array>
+#include <string_view>
+#include <utility>
+
+#include <Kokkos_Macros.hpp>
+
+#ifndef KOKKOS_COMPILER_INTEL
```

Comment:
Doesn't that also exclude the clang based Intel compiler?

Reply:
No, that one defines `KOKKOS_COMPILER_INTEL_LLVM`.